lib/subscribex/publisher/pool.ex

```diff
@@ -35,30 +36,40 @@ defmodule Subscribex.Publisher.Pool do
   def add(broker, pid) do
     broker
     |> group_name
-    |> :pg2.join(pid)
+    |> :pg.join(pid)
   end
 
   # Callback Functions
 
   def init({broker, count, connection_name}) do
-    broker
-    |> group_name
-    |> :pg2.create()
-
     children =
-      for n <- 1..count do
-        worker(
-          Subscribex.Publisher.Pool.Worker,
-          [broker, connection_name],
-          id: :"#{__MODULE__}.Worker.#{n}"
-        )
-      end
+      start_pg() ++
+        for n <- 1..count do
+          %{
+            id: :"#{__MODULE__}.Worker.#{n}",
+            type: :worker,
+            start: {Subscribex.Publisher.Pool.Worker, :start_link, [broker, connection_name]}
+          }
+        end
 
-    supervise(children, strategy: :one_for_one)
+    Supervisor.init(children, strategy: :one_for_one)
   end
 
   # Private Functions
 
+  defp start_pg do
```

Comment:
Is there a better place to start pg for the entire lib? This is so multiple pools dont try to start pg over and over.